scripts/processing/06-make_epochs.py

```diff
@@ -56,51 +56,51 @@ def run_epochs(subject_id, tsss=False):
 
     data_path = op.join(meg_dir, subject)
 
-    # Get all bad channels
-    mapping = map_subjects[subject_id]  # map to correct subject
-    all_bads = list()
+    # map to correct subject for bad channels
+    mapping = map_subjects[subject_id]
+
+    raw_list = list()
+    events_list = list()
+    print("  Loading raw data")
     for run in range(1, 7):
         bads = list()
         bad_name = op.join('bads', mapping, 'run_%02d_raw_tr.fif_bad' % run)
         if os.path.exists(bad_name):
             with open(bad_name) as f:
                 for line in f:
                     bads.append(line.strip())
-        all_bads += [bad for bad in bads if bad not in all_bads]
 
-    raw_list = list()
-    events_list = list()
-    print("  Loading raw data")
-    for run in range(1, 7):
         if tsss:
             run_fname = op.join(data_path, 'run_%02d_filt_tsss_%d_raw.fif'
                                 % (run, tsss))
         else:
             run_fname = op.join(data_path, 'run_%02d_filt_sss_'
                                 'highpass-%sHz_raw.fif' % (run, l_freq))
 
-        raw = mne.io.read_raw_fif(run_fname)
+        raw = mne.io.read_raw_fif(run_fname, preload=True)
 
         delay = int(round(0.0345 * raw.info['sfreq']))
         events = mne.read_events(op.join(data_path, 'run_%02d-eve.fif' % run))
         events[:, 0] = events[:, 0] + delay
         events_list.append(events)
 
-        raw.info['bads'] = all_bads
+        raw.info['bads'] = bads
+        raw.interpolate_bads()
```

Comment:
@jasmainak why did you change this?

Reply:
We have bad channels for each run. So, we should interpolate after loading the `raw` for each run rather than after constructing the epochs